### PR TITLE
Cosmetic - Improve generic parameter usage 

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/MapBasedConfig.java [hunk 1]
```diff
@@ -22,11 +22,10 @@ public MapBasedConfig(Map<String, Object> map) {
         this.map = map;
     }
 
+    @SuppressWarnings("ResultOfMethodCallIgnored")
     public static void clear() {
         File out = new File("target/test-classes/META-INF/microprofile-config.properties");
-        if (out.delete()) {
```
Comment (kenfinnigan) [inline smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/MapBasedConfig.java]: Should this change to do `if (out.exists())` like the change you made in another place?

File: smallrye-reactive-messaging-aws-sns/src/test/java/io/smallrye/reactive/messaging/aws/sns/MapBasedConfig.java [hunk 2]
```diff
@@ -25,6 +25,7 @@ public MapBasedConfig(Map<String, Object> map) {
     public static void clear() {
         File out = new File("target/test-classes/META-INF/microprofile-config.properties");
         if (out.delete()) {
```
Comment (kenfinnigan) [inline smallrye-reactive-messaging-aws-sns/src/test/java/io/smallrye/reactive/messaging/aws/sns/MapBasedConfig.java]: Should this be `out.exists()`?

Comment (kenfinnigan) [reply smallrye-reactive-messaging-aws-sns/src/test/java/io/smallrye/reactive/messaging/aws/sns/MapBasedConfig.java]: There are several places where this if check is made in the code.

In some places the if was removed. in other it still says `out.delete()` and in another, it was changed to `out.exists()`. What should it be?

File: smallrye-reactive-messaging-http/src/main/java/io/smallrye/reactive/messaging/http/converters/AssignationUtils.java [hunk 3]
```diff
@@ -108,6 +102,13 @@ private static boolean isAssignable(Class<?> cls, final Class<?> toClass, final
         return toClass.isAssignableFrom(cls);
     }
 
+    private static boolean isInteger(Class<?> toClass) {
```
Comment (kenfinnigan) [inline smallrye-reactive-messaging-http/src/main/java/io/smallrye/reactive/messaging/http/converters/AssignationUtils.java]: Can this class be replaced with `ClassUtils` from provider artifact?

This check definitely exists in both, maybe more methods are duplicated too?

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/MapBasedConfig.java (hunk 1 above)
Comment (cescoffier) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/MapBasedConfig.java]: should be `isFile`

File: smallrye-reactive-messaging-aws-sns/src/test/java/io/smallrye/reactive/messaging/aws/sns/MapBasedConfig.java (hunk 2 above)
Comment (cescoffier) [reply smallrye-reactive-messaging-aws-sns/src/test/java/io/smallrye/reactive/messaging/aws/sns/MapBasedConfig.java]: Same, it should have been `isFile`. Fixing.

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/MapBasedConfig.java (hunk 1 above)
Comment (kenfinnigan) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/MapBasedConfig.java]: Does this need the check updating or ok to remove completely?

Comment (cescoffier) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/MapBasedConfig.java]: Fixed.

**Pull request merged**